utils.js

```diff
@@ -326,7 +334,7 @@ const syncState = async (allComponents, instance) => {
         await metric.publish()
       }
 
-      promises.push(fn())
+      await fn()
```

Comment:
I think it'll be cleaner (and less confusing for one who may read this code later) to just remove the function wrap in such case, and remove further `await Promise.all()`